build.proj

```diff
@@ -65,7 +65,23 @@
 
   <Import Project="dir.targets" />
   <Import Project="dir.traversal.targets" />
-  <Import Project="$(ToolsDir)clean.targets" />
+  <Import Project="$(ToolsDir)clean.targets" />  
+ 
+  <!-- 
+    This is to initialize the ProjectJsonFiles ItemGroup with the template files during the update dependencies target
+    only to prevent ValidateExactRestore from failing on just the regular project.json files.
+  -->
+  <Target Name="IncludeOnlyTemplateFilesForUpdate" BeforeTargets="UpdateDependencies">
+    <ItemGroup>
+      <ProjectJsonFiles Remove="*" />
```

Comment:
Aha, I thought `ValidateExactRestore` was a target in BuildTools, but it's actually here in build.proj. I agree, no need for the target or for remove--the `ValidateExactRestore` target here should be modified to support the templated project.jsons.